src/plugins/intel_cpu/src/nodes/mkldnn_gather_node.cpp

```diff
@@ -34,79 +39,184 @@ bool MKLDNNGatherNode::isSupportedOperation(const std::shared_ptr<const ov::Node
+        THROW_ERROR << "has incorrect batch_dims " << batchDims << "!";
+
+    dataTypeSize = getOriginalInputPrecisionAtPort(GATHER_DATA).size();
 
-    if (op->get_input_node_shared_ptr(GATHER_AXIS)->get_type_info() == ov::op::v0::Constant::get_type_info_static()) {
+    if (ov::is_type<ov::op::v0::Constant>(op->get_input_node_ptr(GATHER_AXIS))) {
         isAxisInputConst = true;
         axis = ov::as_type<ov::op::v0::Constant>(op->get_input_node_ptr(GATHER_AXIS))->cast_vector<int>()[0];
         if (axis < 0)
             axis += dataSrcRank;
         if (axis < 0 || axis >= dataSrcRank || batchDims > axis)
-            IE_THROW() << errorPrefix << "has incorrect input parameter axis value: " << axis;
+            THROW_ERROR << "has incorrect input parameter axis value: " << axis;
+
+        if (isDataShapeStat) {
+            const auto& dataDims = dataShape.getDims();
+            axisDim = dataDims[axis];
+            beforeAxisSize = std::accumulate(dataDims.begin(), dataDims.begin() + axis, 1lu, std::multiplies<Dim>());
+            betweenBatchAndAxisSize = std::accumulate(dataDims.begin() + batchDims, dataDims.begin() + axis, 1lu, std::multiplies<Dim>());
+            afterAxisSize = std::accumulate(dataDims.begin() + axis + 1, dataDims.end(), 1lu, std::multiplies<Dim>());
+
+            afterAxisSizeInBytes = afterAxisSize * dataTypeSize;
+            axisAndAfterAxisSizeInBytes = axisDim * afterAxisSizeInBytes;
+            srcAfterBatchSizeInBytes = betweenBatchAndAxisSize * axisAndAfterAxisSizeInBytes;
+        }
+    }
+
+    if (isDataShapeStat) {
+        const auto& dataDims = dataShape.getDims();
+        beforeBatchSize = std::accumulate(dataDims.begin(), dataDims.begin() + batchDims, 1lu, std::multiplies<Dim>());
+    }
+    if (isIdxShapeStat) {
+        const auto& idxDims = idxShape.getDims();
+        specIndicesSize = std::accumulate(idxDims.begin() + batchDims, idxDims.end(), 1lu, std::multiplies<Dim>());
+
+        if (isDataShapeStat) {
+            specIdxAndAfterAxSizeB = specIndicesSize * afterAxisSizeInBytes;
+            totalWork = beforeBatchSize * betweenBatchAndAxisSize * specIndicesSize * afterAxisSize;
+        }
     }
-    dataSize = getOriginalInputPrecisionAtPort(GATHER_DATA).size();
 }
 
 void MKLDNNGatherNode::initSupportedPrimitiveDescriptors() {
     if (!supportedPrimitiveDescriptors.empty())
         return;
 
+    auto typeCandidate = ref_any, type = ref_any;
+    size_t vlen = 1;
+    if (x64::mayiuse(x64::avx512_common)) {
+        typeCandidate = jit_avx512;
+        vlen = x64::cpu_isa_traits<x64::avx512_common>::vlen / dataTypeSize;
+    } else if (x64::mayiuse(x64::avx2)) {
+        typeCandidate = jit_avx2;
+        vlen = x64::cpu_isa_traits<x64::avx2>::vlen / dataTypeSize;
+    }
+
+    if (typeCandidate != ref_any) {
+         if (isDynamicNode()) {
+            if (isDataShapeStat && isAxisInputConst && afterAxisSize == 1) {
```

Comment:
This is quite hard restriction that disables JIT impl for most of the cases. Is there any major technical blockers to extend JIT impl applicability for generic case?

Reply:
The execution type is set here for debug purpose. It is redefined in fn ### prepareParams. There is another comment about real supported cases below and description in https://github.com/openvinotoolkit/openvino/pull/6601/files#diff-936f33f6c9b3203891c635a558bf9a050553866c77bea012afa3a0a3345f974bR5